packages/js/src/ui/helpers/createInfiniteScroll.ts

```diff
@@ -0,0 +1,59 @@
+import { Accessor, batch, createComputed, createResource, createSignal, onCleanup, Setter } from 'solid-js';
+import { isServer } from 'solid-js/web';
+
+export function createInfiniteScroll<T>(fetcher: (page: number) => Promise<T[]>): [
+  pages: Accessor<T[]>,
+  loading: Accessor<boolean>,
+  loader: (el: Element) => void,
+  options: {
+    page: Accessor<number>;
+    setPage: Setter<number>;
+    setPages: Setter<T[]>;
+    end: Accessor<boolean>;
+    setEnd: Setter<boolean>;
+  }
+] {
+  const [pages, setPages] = createSignal<T[]>([]);
+  const [initialLoading, setInitialLoading] = createSignal(true);
+  const [page, setPage] = createSignal(0);
```

Comment:
🗒 **note (non-blocking):** the code of this hook would have to be updated after [this PR](https://github.com/novuhq/novu/pull/6086) gets merged